src/canister_tests/src/certificate_validation.rs

```diff
@@ -0,0 +1,145 @@
+// adapted from https://github.com/dfinity/icx-proxy/blob/0cd1a22f717b56ac550a3554a25a845878bfb4e8/src/main.rs#L611
+// TODO: certificate validation should be it's own library
```

Comment:
```suggestion
// TODO: certificate validation should be its own library
```

do we need to track this somewhere?